flightsql/src/main/java/io/deephaven/server/flightsql/FlightSqlResolver.java

```diff
@@ -0,0 +1,1916 @@
+                .submit(() -> ticketHandlerForResolve(session, message));
+
+        // noinspection unchecked
+        return (ExportObject<T>) new TableResolver(ticketHandler, session).submit();
+    }
+
+    private static class TableResolver implements Callable<Table>, Runnable, SessionState.ExportErrorHandler {
+        private final ExportObject<TicketHandler> export;
+        private final SessionState session;
+        private TicketHandler handler;
+
+        public TableResolver(ExportObject<TicketHandler> export, SessionState session) {
+            this.export = Objects.requireNonNull(export);
+            this.session = Objects.requireNonNull(session);
+        }
+
+        public ExportObject<Table> submit() {
+            // We need to provide clean handoff of the Table for Liveness management between the resolver and the
+            // export; as such, we _can't_ unmanage the Table during a call to TicketHandler.resolve, so we must rely
+            // on onSuccess / onError callbacks (after export has started managing the Table).
+            return session.<Table>nonExport()
+                    .require(export)
+                    .onSuccess(this)
+                    .onError(this)
+                    .submit((Callable<Table>) this);
+        }
+
+        // submit
+        @Override
+        public Table call() {
+            handler = export.get();
+            if (!handler.isOwner(session)) {
+                throw new IllegalStateException("Expected TicketHandler to already be authorized for session");
+            }
+            return handler.resolve();
+        }
+
+        // onSuccess
+        @Override
+        public void run() {
+            if (handler == null) {
+                // Should only be run onSuccess, so export.get() must have succeeded
+                throw new IllegalStateException();
+            }
+            release();
+        }
+
+        @Override
+        public void onError(ExportNotification.State resultState, String errorContext, @Nullable Exception cause,
+                @Nullable String dependentExportId) {
+            if (handler == null) {
+                // Will be null if the upstream export has failed
+                return;
+            }
+            release();
+        }
+
+        private void release() {
+            if (!(handler instanceof TicketHandlerReleasable)) {
+                return;
+            }
+            ((TicketHandlerReleasable) handler).release();
+        }
+    }
+
+
+    @Override
+    public <T> SessionState.ExportObject<T> resolve(
+            @Nullable final SessionState session, final Flight.FlightDescriptor descriptor, final String logId) {
+        // This general interface does not make sense; resolution should always be done against a _ticket_. Nothing
+        // calls io.deephaven.server.session.TicketRouter.resolve(SessionState, FlightDescriptor, String)
+        throw new IllegalStateException();
```

Comment:
UnsupportedOperationException? At least add a message of some kind.

Should we drop this slice of the interface from TicketRouter?

Reply:
I think it makes sense to delete this method; after learning more about Flight, I think resolution should always be done from a ticket (in addition to the fact that our code never actually exercises this code path).

I have some further opinions about the role FlightDescriptors + Tickets play after going deep into FlightSQL; it would not be unreasonable for us to consider adopting FlightDescriptor Commands as a way of getting FlightInfo for our various resolvers; afaict, all of our clients assume the ticket structure (as opposed to using our FlightDescriptors Path format) which is _ok_, but if we can have a more officially documented way of constructing FlightDescriptors (whether Paths or Commands), then our tickets really can become opaque.